content/shmem_team_get_options.tex

```diff
@@ -0,0 +1,44 @@
+\apisummary{
+    shmem\_team\_get\_options returns the options flags describing a given team
+}
+
+\begin{apidefinition}
+
+\begin{Csynopsis}
+long @\FuncDecl{shmem\_team\_get\_options}@(shmem_team_t team);
+\end{Csynopsis}
+
+\begin{apiarguments}
+\apiargument{IN}{team}{A valid SHMEM team handle.}
+\end{apiarguments}
+
+\apidescription{
+\FUNC{shmem\_team\_get\_options} returns a long unsigned value containing
+all of the options which describe the given team. Options are requested when
+new teams are created in the various \FUNC{shmem\_team\_split\_*} functions.
+Whichever of the requested options are applied to the team by the library
```

Comment:
> Whichever of the requested options are applied to the team...

suggest:

> All of the requested options applied to the team...

Does this sentence imply the library may apply *more* options than what is requested by the call to `shmem_team_split_*`?

Reply:
Good question. I would think that generally, we don't want to allow implementations to apply non-default options to teams, since this would hinder portability. And, at this time, SHMEM_TEAM_WORLD and SHMEM_TEAM_NODE are specified to always have options = 0, i.e. all default options.

However, I remember that there has been some discussion around allowing programmatic initialization of SHMEM_TEAM_WORLD for the purpose of applying different heaps or memory type options. So, I would suggest that until or unless SHMEM_TEAM_WORLD can have some non-default options applied to it, then the implementation should not be allowed to apply any non-requested options to a new team. Basically, there ought to be a mechanism for the program to block or apply team options globally before some implementation can start applying options to teams other than what is specified in the team split call.